### PR TITLE
Upgraded project to VS2017

### LINKED ISSUE #325: Upgrade project to VS2017
**I'm submitting a ...**  (check one with "x")

- [ ] bug report => search github for a similar issue or PR before submitting.                          
- [x] feature request => 

**Step/s to reproduce issue**
Upgrade project from xproj to csproj to support upgrading to latest version of DotNet core

**Screen shot**

### PR DESCRIPTION
**Pull request by juliepromact**
Fixes #325 

**Files changed are:**

**1. .travis.yml -**  Updated version and path.
**2.  Dockerfile -** Updated version and removed references to project,json.
**3. Promact.Oauth.Server.sln -** Visual Studio version updated.
**4. global.json -** Removed.
**5. Promact.Oauth.Server.Tests.csproj -** Added.
**6. Promact.Oauth.Server.Tests.xproj -** Removed.
**7. project.json -** Removed.
**8. Promact.Oauth.Server.csproj -** Added
**9. Promact.Oauth.Server.xproj -** Removed
**10. Promact.Oauth.Server/project.json -** Removed.
**11. appveyor.yml -** Updated version and path.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,8 @@
-FROM promact/docker-dotnet-nodejs:latest
+FROM promact/docker-dotnet-nodejs:1.1.0-sdk-msbuild
```
Comment (chintans) [inline Dockerfile]: This will not work. keep it as latest

Comment (juliepromact) [inline Dockerfile]: Ok will update it.

File: Promact.Oauth.Server/src/Promact.Oauth.Server/project.json [hunk 2]
```diff
@@ -1,116 +0,0 @@
...
-    "preserveCompilationContext": true
-  },
-
-  "runtimeOptions": {
-    "configProperties": {
-      "System.GC.Server": true
-    }
-  },
-
-  "publishOptions": {
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/project.json]: How will publishOptions work now since project.json is removed

Comment (juliepromact) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/project.json]: In .csproj

<ItemGroup>
    <None Update="wwwroot\**\*;Views\**\*;Areas\**\Views;nlog.config;Template\**\*">
      <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
    </None>
  </ItemGroup>

the above given code has been generated.

Comment (juliepromact) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/project.json]: I have updated it to 
`
<ItemGroup>
    <None Update="wwwroot\**\*;Views\**\*;Areas\**\Views;nlog.config;Template\**\*;appsettings.json;StringLiterals\stringliterals.json;Seed\seeddata.json">
      <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
    </None>
  </ItemGroup>
`

**Pull request merged**